src/OpenTelemetry/AssemblyInfo.cs

```diff
@@ -18,10 +18,8 @@
 
 [assembly: InternalsVisibleTo("OpenTelemetry.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Exporter.InMemory" + AssemblyInfo.PublicKey)]
-[assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Prometheus.Shared" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Prometheus.AspNetCore" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Prometheus.AspNetCore.Tests" + AssemblyInfo.PublicKey)]
-[assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Prometheus.HttpListener" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Prometheus.HttpListener.Tests" + AssemblyInfo.PublicKey)]
```

Comment:
Can we remove tests too or no?

Reply:
Tried and they have dependency on `MeterProvider.TryFindExporter`.

Reply:
This is the test code which makes use of the internals https://github.com/open-telemetry/opentelemetry-dotnet/blob/a8c222ab5465a1612f1bf3865feb7d5e601696d8/test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusCollectionManagerTests.cs#L51

This is where the internal method is defined https://github.com/open-telemetry/opentelemetry-dotnet/blob/a8c222ab5465a1612f1bf3865feb7d5e601696d8/src/OpenTelemetry/Metrics/MeterProviderExtensions.cs#L116

Reply:
Could probably have a reflection-based alternative for tests.

Looks like PrometheusExporterMiddleware also uses this.